src/main/kotlin/utils/CoverageFileStatistics.kt

```diff
@@ -0,0 +1,64 @@
+package utils
+import org.xml.sax.Attributes
+import org.xml.sax.InputSource
+import org.xml.sax.helpers.DefaultHandler
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import javax.xml.parsers.SAXParserFactory
+import kotlin.math.min
+
+object CoverageFileStatistics {
+
+    private class Handler() : DefaultHandler() {
+
+        var currentPrefix = ""
+        var coveredMethods = 0
+        var missedMethods = 0
+        var coveredByPrefix = HashMap<String, Int>()
+        var missedByPrefix = HashMap<String, Int>()
+
+        override fun resolveEntity(publicId: String?, systemId: String?): InputSource {
+            return InputSource(ByteArrayInputStream(byteArrayOf()))
+        }
+
+        override fun startElement(uri: String?, localName: String?, qName: String, attributes: Attributes) {
+            when (qName) {
+                "counter" -> {
+                    if (attributes.getValue("type") == "METHOD") {
+                        coveredMethods = Integer.valueOf(attributes.getValue("covered"))
+                        missedMethods = Integer.valueOf(attributes.getValue("missed"))
+                    }
+                }
+                "package" -> {
+                    val currentPackageName = attributes.getValue("name")?.replace("/".toRegex(), ".") ?: ""
+                    val parts = currentPackageName.split(".")
+                    currentPrefix = parts.subList(0, min(parts.size, 3)).joinToString(".")
```

Comment:
please extract the package mangling logic into a function and document what this does. It seems rather "magic" with the magic number 3 in there and lots of in-line string splitting and joining
I think future readers might wonder what the purpose of this is

Reply:
also, when extracting the function, you can use

  ... ?: return ""

in line 33 instead of

  ... ?: ""

to make it clearer for readers what will be returned if the attribute is missing